### PR TITLE
[JENKINS-36289] Added ReplayAction#run2

### PR DESCRIPTION
**Pull request by imeredith**
Added run2() to ReplayAction which returns Queue.Item rather than
QueueTaskFuture, so that Blue Ocean can reference the item in the
queue for its REST api.

Note: I'm not sure what the best way to do this was so I added run2 rather than modifying run's return signature.

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java [hunk 1]
```diff
@@ -165,6 +166,18 @@ public void doRun(StaplerRequest req, StaplerResponse rsp) throws ServletExcepti
      * @return a way to wait for the replayed build to complete
      */
     public @CheckForNull QueueTaskFuture/*<Run>*/ run(@Nonnull String replacementMainScript, @Nonnull Map<String,String> replacementLoadedScripts) {
+        Queue.Item item = run2(replacementMainScript, replacementLoadedScripts);
+        return item == null ? null : item.getFuture();
+    }
+
+    /**
+     * For use in projectis that want initiate a replay via the Java API.
```
Comment (michaelneale) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java]: projectis/project?

**Comment by michaelneale** [comment]
Seems like a needed feature - I wonder if "run2" can have a better name (really the return type gives the purpose away, which is something people will see when they use content-assist in an IDE and type "run.." so having the method start with run makes sense to me.

**Comment by michaelneale** [comment]
🐝  pending 🐜  for javadoc typo

**Comment by michaelneale** [comment]
@imeredith this needs a corresponding JIRA ticket

File: src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java]: :ant: `projects` I think was meant.

**Comment by jglick** [comment]
The name is a little weird but good enough. :bee:

**Comment by michaelneale** [comment]
@jglick I propose win32 api conventions of runEx runExA runEx2A etc ;)

**Pull request merged**